### PR TITLE
Fix filtered data lag in plots

### LINKED ISSUE #132: Median Filter Time offset for certain datasets
For a particularly noisy dataset, the median filter seems to have gotten the timestamps shifted.

image

### PR DESCRIPTION
**Pull request by Piphi5**
Resolves #132 

Uses Timestamp instead of dts to plot the data.

Also fixes some additional bugs in the current analysis pipeline.

image

### CODE REVIEW

File: src/main/native/cpp/analysis/FilteringUtils.cpp [hunk 1]
```diff
@@ -10,7 +10,11 @@
 #include <frc/LinearFilter.h>
 #include <frc/MedianFilter.h>
 #include <units/math.h>
+#include <wpi/StringRef.h>
 #include <wpi/math>
+#include <wpi/raw_ostream.h>
```
Comment (calcmogul) [inline src/main/native/cpp/analysis/FilteringUtils.cpp:15]: Can this include be removed?

File: src/main/native/cpp/analysis/FilteringUtils.cpp [hunk 2]
```diff
@@ -10,7 +10,11 @@
...
+
+#include "sysid/analysis/Storage.h"
```
Comment (calcmogul) [inline src/main/native/cpp/analysis/FilteringUtils.cpp:17]: This include shouldn't be needed because it's already included by FilteringUtils.h.

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Comment by calcmogul** [comment]
The quasistatic sim with that data misbehaves a bit. Is the higher velocity threshold what's causing it? Perhaps the low velocity threshold is hiding a bug.
Screenshot_2021-05-08_19-59-45

File: src/main/native/cpp/analysis/FilteringUtils.cpp [hunk 3]
```diff
@@ -10,7 +10,11 @@
 #include <frc/LinearFilter.h>
 #include <frc/MedianFilter.h>
 #include <units/math.h>
+#include <wpi/StringRef.h>
```
Comment (calcmogul) [inline src/main/native/cpp/analysis/FilteringUtils.cpp:13]: StringRef is used in FilteringUtils.h as well, so this include should be moved there.

**Comment by Piphi5** [comment]
> The quasistatic sim with that data misbehaves a bit. Is the higher velocity threshold what's causing it? Perhaps the low velocity threshold is hiding a bug.

It seems like the kV gains are pretty bad with the low motion threshold. But the lower velocity threshold could be causing some issues as well

**Review by calcmogul — APPROVED** [review]

**Pull request merged**